### PR TITLE
ci: add aws edge-ami cleanup job

### PR DESCRIPTION
**Pull request by mcattamoredhat**
This PR cleans some of the idle AWS service elements in the AWS Default Region.

`aws-ami-cleanup.sh` script searches and removes buckets, images, snapshots, and key pairs that have been created more than 4 hours ago at the workflow triggering moment.

`cleanup-aws-ami.yaml` workflow allows both manual and scheduled triggering of the script (every four hours).

### CODE REVIEW

File: tools/aws-ami-cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+           aws s3 rb "s3://${BUCKET_NAME}" --force > /dev/null
+        fi
+    done
+fi
+
+# List Registered Images
+greenprint "Checking AWS EC2 Idle AMIs 💽"
+aws ec2 describe-images \
+    --owners self \
+    --query 'Images[?(Tags[?Value == `composer-ci`].Value)]' \
```
Comment (henrywang) [inline tools/aws-ami-cleanup.sh:72]: Is the `'` required? Can it be changed to `"`? Thanks.

**Review by henrywang — CHANGES_REQUESTED** [review]

File: .github/workflows/cleanup-aws-ami.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+name: AWS edge ami cleanup job
+
+on:
+  workflow_dispatch:
+  schedule:
```
[the remark this answers is not in the document]
Comment (mcattamoredhat) [reply .github/workflows/cleanup-aws-ami.yaml:6]: Done, thanks!

File: .github/workflows/cleanup-aws-ami.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    # Run at 10:00 every day
+    - cron: '0 10 * * *'
+    # Run at 22:00 every day
+    - cron: '0 22 * * *'
+
+jobs:
+  cleanup-ami:
+    runs-on: [kite, x86_64, rhos-01, rhel-9-3, medium]
```
[the remark this answers is not in the document]
Comment (mcattamoredhat) [reply .github/workflows/cleanup-aws-ami.yaml:14]: Done, thanks!

File: tools/aws-ami-cleanup.sh (hunk 1 above)
Comment (mcattamoredhat) [reply tools/aws-ami-cleanup.sh:72]: If we decide to keep single quotes, then we will need to land [PR#4210](https://github.com/virt-s1/rhel-edge/pull/4210)

Comment (mcattamoredhat) [reply tools/aws-ami-cleanup.sh:77]: Another workaround has been added. It is just to ignore Shellcheck SC2016.

File: .github/workflows/cleanup-aws-ami.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+      - name: Cleanup aws resources
+        run: ./tools/aws-ami-cleanup.sh
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: us-east-1
+          DOWNLOAD_NODE: ${{ secrets.DOWNLOAD_NODE }}
```
Comment (henrywang) [inline .github/workflows/cleanup-aws-ami.yaml:22]: Thanks @mcattamoredhat. `DOWNLOAD_NODE` is not required in this case.

**Review by henrywang — CHANGES_REQUESTED** [review]

File: .github/workflows/cleanup-aws-ami.yaml (hunk 4 above)
Comment (mcattamoredhat) [reply .github/workflows/cleanup-aws-ami.yaml]: Done, thanks!

**Review by henrywang — APPROVED** [review]

**Comment by henrywang** [comment]
Let's wait for all green.

**Pull request merged**